openapi-cli/src/main/java/io/ballerina/openapi/generators/client/mime/MultipartFormData.java

```diff
@@ -0,0 +1,177 @@
+import static io.ballerina.openapi.generators.GeneratorConstants.BALLERINA;
+import static io.ballerina.openapi.generators.GeneratorConstants.MIME;
+import static io.ballerina.openapi.generators.GeneratorUtils.checkImportDuplicate;
+import static io.ballerina.openapi.generators.GeneratorUtils.escapeIdentifier;
+import static io.ballerina.openapi.generators.GeneratorUtils.getValidName;
+
+/**
+ * Defines the payload structure of multipart form-data mime type.
+ */
+public class MultipartFormData extends MimeType {
+
+    BallerinaUtilGenerator ballerinaUtilGenerator;
+    List<ImportDeclarationNode> imports;
+
+    public MultipartFormData(List<ImportDeclarationNode> imports,
+                             BallerinaUtilGenerator ballerinaUtilGenerator) {
+        this.imports = imports;
+        this.ballerinaUtilGenerator = ballerinaUtilGenerator;
+    }
+
+    @Override
+    public void setPayload(List<StatementNode> statementsList, Map.Entry<String, MediaType> mediaTypeEntry) {
+        ballerinaUtilGenerator.setRequestBodyMultipartFormDatafound(true);
+        ImportDeclarationNode mimeImport = GeneratorUtils.getImportDeclarationNode(BALLERINA, MIME);
+        if (!checkImportDuplicate(imports, MIME)) {
+            imports.add(mimeImport);
+        }
+        VariableDeclarationNode encodingMap = getMultipartMap(mediaTypeEntry);
+
+        VariableDeclarationNode bodyPartsVariable;
+        if (encodingMap == null) {
+            bodyPartsVariable = GeneratorUtils.getSimpleStatement("mime:Entity[]", "bodyParts",
+                    "check createBodyParts(payload)");
+        } else {
+            statementsList.add(encodingMap);
+            bodyPartsVariable = GeneratorUtils.getSimpleStatement("mime:Entity[]", "bodyParts",
+                    "check createBodyParts(payload, encodingMap)");
+        }
+
+        statementsList.add(bodyPartsVariable);
+        ExpressionStatementNode setPayloadExpression = GeneratorUtils.getSimpleExpressionStatementNode(
+                "request.setBodyParts(bodyParts)");
+        statementsList.add(setPayloadExpression);
+    }
+
+    public VariableDeclarationNode getMultipartMap(Map.Entry<String, MediaType> mediaTypeEntry) {
+        List<Node> filedOfMap = new ArrayList<>();
+        BuiltinSimpleNameReferenceNode mapType = createBuiltinSimpleNameReferenceNode(null,
+                createIdentifierToken("map<Encoding>"));
+        CaptureBindingPatternNode bindingPattern = createCaptureBindingPatternNode(
+                createIdentifierToken("encodingMap"));
+        TypedBindingPatternNode bindingPatternNode = createTypedBindingPatternNode(mapType, bindingPattern);
+
+        if (mediaTypeEntry.getValue().getEncoding() != null) {
+            Set<Map.Entry<String, Encoding>> endcodingEntries = mediaTypeEntry.getValue().getEncoding().entrySet();
+            for (Map.Entry<String, Encoding> entry : endcodingEntries) {
+                SpecificFieldNode contentField = null;
```

Comment:
I would rather make a separate function with the logic insiide this `for` loop for readability.

Reply:
Addressed in https://github.com/ballerina-platform/openapi-tools/commit/b073cda18ba34b0a9e050f7cc820da3107ffa33b